core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -245,158 +301,30 @@ struct MDRangePolicy : public Kokkos::Impl::PolicyTraits<Properties...> {
-    else  // ROCm
-    {
-      index_type span;
-      int increment  = 1;
-      int rank_start = 0;
-      int rank_end   = rank;
-      if ((int)inner_direction == (int)Right) {
-        increment  = -1;
-        rank_start = rank - 1;
-        rank_end   = -1;
-      }
-      for (int i = rank_start; i != rank_end; i += increment) {
-        span = m_upper[i] - m_lower[i];
-        if (m_tile[i] <= 0) {
-          // TODO: determine what is a good default tile size for rocm
-          // may be rank dependent
-          if (((int)inner_direction == (int)Right && (i < rank - 1)) ||
-              ((int)inner_direction == (int)Left && (i > 0))) {
-            if (m_prod_tile_dims < 256) {
-              m_tile[i] = 4;
-            } else {
-              m_tile[i] = 1;
-            }
-          } else {
-            m_tile[i] = 16;
-          }
-        }
-        m_tile_end[i] =
-            static_cast<index_type>((span + m_tile[i] - 1) / m_tile[i]);
-        m_num_tiles *= m_tile_end[i];
-        m_prod_tile_dims *= m_tile[i];
-      }
-      if (m_prod_tile_dims > 1024) {  // but product num_threads < 1024
-        printf(" Tile dimensions exceed ROCm limits\n");
-        Kokkos::abort(
-            " ROCm ExecSpace Error: MDRange tile dims exceed maximum number of "
-            "threads per block - choose smaller tile dims");
-        // Kokkos::Impl::throw_runtime_exception( " Cuda ExecSpace Error:
-        // MDRange tile dims exceed maximum number of threads per block - choose
-        // smaller tile dims");
-      }
-    }
-#endif
+  // </editor-fold> end Constructors, destructor, and assignment }}}2
+  //----------------------------------------------------------------------------
+
+  KOKKOS_INLINE_FUNCTION const execution_space& space() const {
+    return m_space;
   }
 
-  template <typename LT, typename UT, typename TT = array_index_type>
-  void init(std::initializer_list<LT> const& lower,
-            std::initializer_list<UT> const& upper,
-            std::initializer_list<TT> const& tile = {}) {
-    if (static_cast<int>(m_lower.size()) != rank ||
-        static_cast<int>(m_upper.size()) != rank)
-      Kokkos::abort(
-          "MDRangePolicy: Constructor initializer lists have wrong size");
-
-    for (auto i = 0; i < rank; ++i) {
-      m_lower[i] = static_cast<array_index_type>(lower.begin()[i]);
-      m_upper[i] = static_cast<array_index_type>(upper.begin()[i]);
-      if (static_cast<int>(tile.size()) == rank)
-        m_tile[i] = static_cast<array_index_type>(tile.begin()[i]);
-      else
-        m_tile[i] = 0;
+ private:
+  // TODO this doesn't need to be replicated for every MDRangePolicy
+  //      specialization; consider moving it to a free function or something
+  static _point_type _get_narrowed_values(point_type const& a) noexcept {
+    // We don't have access to std::integer_sequence, so we have to do this
+    _point_type rv = {/* workaround for nvcc compiler/linker bug */ 0};
```

Comment:
What's the deal with that?  Default constructed `rv` is what gave you the bug?  How did aggregate initialization of the array with a single zero entry regardless of the rank help?

Reply:
I **think** there's an issue with aggregate construction in `nvcc` that was causing symbols for default constructors to fail to be generated when `{ }` is given as the aggregate initializer.  That's the only thing I can think of that would lead to this being a fix.